[content-visibility] Improve content-visibility: auto code

rwlbuis opened the pull request:
#### 6f118d4b6dd04b9ffcb2162d641fff1242e0c804
<pre>
[content-visibility] Improve content-visibility: auto code
<a href="https://bugs.webkit.org/show_bug.cgi?id=236711">https://bugs.webkit.org/show_bug.cgi?id=236711</a>

Reviewed by Simon Fraser.

Address review comments from r266084.

* Source/WebCore/dom/ContentVisibilityDocumentState.cpp:
(WebCore::ContentVisibilityDocumentState::unobserve):
(WebCore::ContentVisibilityDocumentState::updateRelevancyOfContentVisibilityElements):
(WebCore::ContentVisibilityDocumentState::updateContentRelevancyForScrollIfNeeded):
(WebCore::ContentVisibilityDocumentState::updateContentRelevancyStatusForScrollIfNeeded): Deleted.
* Source/WebCore/dom/ContentVisibilityDocumentState.h:
* Source/WebCore/dom/Document.cpp:
(WebCore::Document::setFocusedElement):
(WebCore::Document::updateRelevancyOfContentVisibilityElements):
(WebCore::Document::scheduleContentRelevancyUpdate):
(WebCore::Document::updateContentRelevancyForScrollIfNeeded):
* Source/WebCore/dom/Document.h:
* Source/WebCore/dom/Element.cpp:
(WebCore::Element::scrollIntoView):
(WebCore::Element::scrollIntoViewIfNeeded):
(WebCore::Element::focus):
(WebCore::Element::addToTopLayer):
(WebCore::Element::removeFromTopLayer):
(WebCore::Element::isRelevantToUser const):
(WebCore::Element::contentRelevancy const):
(WebCore::Element::setContentRelevancy):
(WebCore::Element::contentVisibilityViewportChange):
(WebCore::Element::contentRelevancyStatus const): Deleted.
(WebCore::Element::setContentRelevancyStatus): Deleted.
* Source/WebCore/dom/Element.h:
* Source/WebCore/dom/ElementRareData.cpp:
* Source/WebCore/dom/ElementRareData.h:
(WebCore::ElementRareData::contentRelevancy const):
(WebCore::ElementRareData::setContentRelevancy):
(WebCore::ElementRareData::contentRelevancyStatus const): Deleted.
(WebCore::ElementRareData::setContentRelevancyStatus): Deleted.
* Source/WebCore/page/LocalFrameView.cpp:
(WebCore::LocalFrameView::maintainScrollPositionAtAnchor):
* Source/WebCore/page/Page.h:

Canonical link: <a href="https://commits.webkit.org/267130@main">https://commits.webkit.org/267130@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/bc0ad01b3e35aa8a9e0b23fa2ae8628271bea0fc

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/15774 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/26/builds/16092 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/16469 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/17536 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/14804 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/15959 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/23/builds/19090 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/16/builds/16188 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/17299 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/15964 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/15/builds/16434 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/13429 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/18291 "Built successfully") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/39/builds/13682 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/14241 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/21144 "Passed tests") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/14682 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/14406 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/17664 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/7/builds/15003 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/17/builds/12717 "Passed tests") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/30/builds/14250 "Built successfully") | | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/3762 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/4/builds/18619 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/31/builds/14825 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

smfr commented on Source/WebCore/dom/Document.cpp:
```diff
@@ -9594,16 +9594,24 @@ void Document::updateRelevancyOfContentVisibilityElements()
     if (!isObservingContentVisibilityTargets())
         return;
-    m_contentRelevancyStatusUpdate.add(status);
+    m_contentRelevancyUpdate.add(contentRelevancy);
```
Should this also schedule a rendering update? The "schedule" in the name isn't quite right if it just sets a bit.

rwlbuis replied:
Yes, that makes sense, added it.

nt1m commented on Source/WebCore/dom/ContentVisibilityDocumentState.cpp:
```diff
@@ -94,24 +94,24 @@ IntersectionObserver* ContentVisibilityDocumentState::intersectionObserver(Docum
+DidUpdateAnyContentRelevancy ContentVisibilityDocumentState::updateRelevancyOfContentVisibilityElements(const OptionSet<ContentRelevancy>& relevancyToCheck)
 {
-    bool didUpdateAnyContentRelevancy = false;
+    DidUpdateAnyContentRelevancy didUpdateAnyContentRelevancy = DidUpdateAnyContentRelevancy::No;
```
```suggestion
    auto didUpdateAnyContentRelevancy = DidUpdateAnyContentRelevancy::No;
```

rwlbuis replied:
Sure, done.

rwlbuis committed:
[content-visibility] Improve content-visibility: auto code
https://bugs.webkit.org/show_bug.cgi?id=236711

Reviewed by Simon Fraser.

Address review comments from r266084.

* Source/WebCore/dom/ContentVisibilityDocumentState.cpp:
(WebCore::ContentVisibilityDocumentState::unobserve):
(WebCore::ContentVisibilityDocumentState::updateRelevancyOfContentVisibilityElements):
(WebCore::ContentVisibilityDocumentState::updateContentRelevancyForScrollIfNeeded):
(WebCore::ContentVisibilityDocumentState::updateContentRelevancyStatusForScrollIfNeeded): Deleted.
* Source/WebCore/dom/ContentVisibilityDocumentState.h:
* Source/WebCore/dom/Document.cpp:
(WebCore::Document::setFocusedElement):
(WebCore::Document::updateRelevancyOfContentVisibilityElements):
(WebCore::Document::scheduleContentRelevancyUpdate):
(WebCore::Document::updateContentRelevancyForScrollIfNeeded):
* Source/WebCore/dom/Document.h:
* Source/WebCore/dom/Element.cpp:
(WebCore::Element::scrollIntoView):
(WebCore::Element::scrollIntoViewIfNeeded):
(WebCore::Element::focus):
(WebCore::Element::addToTopLayer):
(WebCore::Element::removeFromTopLayer):
(WebCore::Element::isRelevantToUser const):
(WebCore::Element::contentRelevancy const):
(WebCore::Element::setContentRelevancy):
(WebCore::Element::contentVisibilityViewportChange):
(WebCore::Element::contentRelevancyStatus const): Deleted.
(WebCore::Element::setContentRelevancyStatus): Deleted.
* Source/WebCore/dom/Element.h:
* Source/WebCore/dom/ElementRareData.cpp:
* Source/WebCore/dom/ElementRareData.h:
(WebCore::ElementRareData::contentRelevancy const):
(WebCore::ElementRareData::setContentRelevancy):
(WebCore::ElementRareData::contentRelevancyStatus const): Deleted.
(WebCore::ElementRareData::setContentRelevancyStatus): Deleted.
* Source/WebCore/page/LocalFrameView.cpp:
(WebCore::LocalFrameView::maintainScrollPositionAtAnchor):
* Source/WebCore/page/Page.h:

Canonical link: https://commits.webkit.org/267130@main
